nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -364,7 +364,7 @@ in
         # If the empty string is assigned to this option, the list of commands to start is reset, prior assignments of this option will have no effect.
         ExecStart = [ "" ''${cfg.package}/bin/caddy run ${runOptions} ${optionalString cfg.resume "--resume"}'' ];
         # Validating the configuration before applying it ensures we’ll get a proper error that will be reported when switching to the configuration
-        ExecReload = [ "" ''${cfg.package}/bin/caddy reload ${runOptions} --force'' ];
+        ExecReload = [ "" ] ++ lib.optional cfg.enableReload ''${cfg.package}/bin/caddy reload ${runOptions} --force'';
```

Comment:
```suggestion
        ExecReload = lib.optionals cfg.enableReload [ ''${lib.getExe cfg.package} reload ${runOptions} --force'' ];
```

Reply:
No, the empty list item is explicitly there to clear the list from any previous entries (see systemd man pages).

If we apply the change you proposed, enableDefault set to false would make the config in the upstream unit file shine through.